### DISCUSSION TITLE
Serving both HTTP on port x and HTTPS on port y at the same time?

### DISCUSSION THREAD

**Discussion by josb** [tokio-rs/axum#1066] [Q&A] [answered]
```
// HTTP
 let server_http : Server<AddrIncoming, IntoMakeService<Router>> = axum::Server::try_bind(&socket_http)?.serve(endpoints::create_app(service_http).into_make_service());
// HTTPS
 let server_https : Server<AddrIncoming, IntoMakeService<Router>> = axum::Server::???
```
What is the equivalent code for supporting HTTPS, such that both `server_http` and `server_https` have the same type? This is needed so that both can be pushed onto a `Vec` which can then be passed as a parameter to `futures::future::try_join_all()`.

**Comment by davidpdrsn** [comment, marked as answer]
axum doesn't have built in support for serving via https but it can be implemented in you app or by a library. Take a look at the axum-server crate.

**Reply by josb** [reply]
Thank you, David, for pointing me in the right direction. Meanwhile I've been able to make this work as you suggested.
